From aa82809640f8ffdf69030cd32f60300840ead80a Mon Sep 17 00:00:00 2001
From: Zhang Qiang <qiang.z.zhang@intel.com>
Date: Tue, 3 Dec 2013 17:24:09 +0800
Subject: [PATCH 9/9] always register qemu handler

always register qemu handler even if interpreter does not exist, and
in this case report warning instead of skip it. This way it just work
like qemu-binfmt-conf.sh script in qemu

Change-Id: Idf0f5eb8b6a4fec53ea272d2424beb9658cec6a4
---
 initvm.c |    7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/initvm.c b/initvm.c
index bed0cc4..dd2495c 100644
--- a/initvm.c
+++ b/initvm.c
@@ -212,10 +212,9 @@ enum okfail binfmt_register(char *datafile, char *regfile)
 
 
 		if (access(f[interpreter], X_OK) != 0) {
-			fprintf(stderr, 
-				"%s: line %d: interpreter '%s' not found,"
-				" ignoring\n", datafile, line, f[interpreter]);
-			goto skip;
+			fprintf(stderr,
+				"warning: %s: line %d: interpreter '%s' not "
+				"found\n", datafile, line, f[interpreter]);
 		}
 
 		if (!write_file_string(regfile, buf)) {
-- 
1.7.9.5

